File: Aula 14 - Resiliência e persistencia/logico.py/nivel3.py
try:

    arquivo = open("dados_financeiros.json", "r")

    conteudo = arquivo.read()

    print(conteudo)

    arquivo.close()

except FileNotFoundError:

    print("Arquivo não encontrado.")

    with open("dados_financeiros.json", "a") as arquivo:

        arquivo.write("{ }\n")

        print("Arquivo criado automaticamente.")

File: Aula 14 - Resiliência e persistencia/logico.java/nivel3.java
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.util.Scanner;

public class Nivel3 {

    public static void main(String[] args) {

        try {

            File arquivo = new File("dados_financeiros.json");

            Scanner leitor = new Scanner(arquivo);

            while(leitor.hasNextLine()) {

                System.out.println(leitor.nextLine());

            }

            leitor.close();

        }

        catch(IOException erro) {

            System.out.println("Arquivo não encontrado.");

            try {

                FileWriter arquivoNovo = new FileWriter("dados_financeiros.json");

                arquivoNovo.write("{ }\n");

                arquivoNovo.close();

                System.out.println("Arquivo criado automaticamente.");

            }

            catch(IOException e) {

                System.out.println("Erro ao criar arquivo.");

            }

        }

    }
}